src/libraries/shims/src/mscorlib.csproj

```diff
@@ -7,4 +7,11 @@
   <ItemGroup>
     <Compile Include="mscorlib.forwards.cs" />
   </ItemGroup>
+  <Target Name="RemoveConflictingAssemblyReference" AfterTargets="FindReferenceAssembliesForReferences">
+    <ItemGroup>
+      <!-- System.Collections.Generic.CollectionExtensions type exists in both System.Private.CoreLib and Microsoft.Extensions.DependencyModel, we only want the former -->
+      <ReferencePathWithRefAssemblies Remove="@(ReferencePathWithRefAssemblies)"
+                                      Condition="'%(Filename)' == 'Microsoft.Extensions.DependencyModel'" />
```

Comment:
Use item functions instead of batching as it performs better:

```suggestion
      <!-- System.Collections.Generic.CollectionExtensions type exists in both System.Private.CoreLib and Microsoft.Extensions.DependencyModel, we only want the former -->
      <ReferencePathWithRefAssemblies Remove="@(ReferencePathWithRefAssemblies->WithMetadataValue('Filename', 'Microsoft.Extensions.DependencyModel'))" />
```